packages/amplify-cli/src/lib/migrate-project.js

```diff
@@ -0,0 +1,188 @@
+      // This is an older project & migration is needed
+      generateNewProjectConfig(projectConfig, projectConfigFilePath);
+      generateLocalEnvInfo(projectConfig);
+      generateAwsLocalInfo();
+      generateTeamProviderInfo();
+      generateBackendConfig();
+      generateGitIgnoreFile();
+    }
+
+    // Give each category a chance to migrate their respective files
+
+    const categoryMigrationTasks = [];
+    const amplifyMetafilePath = getAmplifyMetaFilePath();
+    const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+    const initializedCategories = Object.keys(amplifyMeta);
+
+    const categoryPlugins = {};
+    plugins.forEach((plugin) => {
+      if (plugin.name.includes('category')) {
+        const strs = plugin.name.split('-');
+        const categoryName = strs[strs.length - 1];
+        categoryPlugins[categoryName] = plugin.directory;
+      }
+    });
+
+    const availableCategory = Object.keys(plugins).filter(key =>
+      initializedCategories.includes(key));
+
+    availableCategory.forEach((category) => {
+      try {
+        const { migrateCategoryFiles } = require(categoryPlugins[category]);
+        if (migrateCategoryFiles) {
+          categoryMigrationTasks.push(() => migrateCategoryFiles());
+        }
+      } catch (e) {
+        print.warning(`Could not run migrateProject function for ${category} category`);
+      }
+    });
+
+    spinner.start('Migrating your project');
+    await Promise.all(categoryMigrationTasks);
+    spinner.succeed('Migrated your project successfully.');
+  } catch (e) {
+    spinner.fail('There was an error migrating your project.');
+    throw e;
+  }
+}
+
+function generateNewProjectConfig(projectConfig, projectConfigFilePath) {
+  const newProjectConfig = {};
+
+  Object.assign(newProjectConfig, projectConfig);
+  // These attributes are now stores in amplify/.config/local-env-info.json
+  delete newProjectConfig.projectPath;
+  delete newProjectConfig.defaultEditor;
+
+  // Modify frontend handler
+  const frontendPluginPath = Object.keys(projectConfig.frontendHandler)[0];
+  const frontendPlugin = frontendPluginPath.split('/')[frontendPluginPath.split('/').length - 1];
+  const frontend = frontendPlugin.split('-')[frontendPlugin.split('-').length - 1];
+
+  newProjectConfig.frontend = frontend;
+  delete newProjectConfig.frontendHandler;
+
+  // Modify provider handler
+  const provider = Object.keys(projectConfig.providers)[0];
+  newProjectConfig.providers = [provider];
+
+  const jsonString = JSON.stringify(newProjectConfig, null, 4);
+  fs.writeFileSync(projectConfigFilePath, jsonString, 'utf8');
+}
+
+function generateLocalEnvInfo(projectConfig) {
+  const envInfo = {
+    projectPath: projectConfig.projectPath,
+    defaultEditor: projectConfig.defaultEditor,
+    envName: 'NONE',
```

Comment:
Should we ask the user what would they like to name this env instead of naming it none?

Reply:
Don't think so. We've changed our CFN files accordingly.

Reply:
Internally it would be none. But for the user it could be `prod` env or `dev` env